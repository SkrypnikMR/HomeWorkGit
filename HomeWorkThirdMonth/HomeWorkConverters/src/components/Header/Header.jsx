import React from 'react';
import { StHeaderDiv } from './styled';

function Header() {
    return (
        <StHeaderDiv><h1>Converter</h1></StHeaderDiv>
    );
}

export default Header;
