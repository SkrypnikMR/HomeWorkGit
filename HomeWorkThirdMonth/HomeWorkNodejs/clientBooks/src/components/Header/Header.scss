header {
    width: 100%;
    height: 100px;
    border: 1px solid black;
    background-color: #4A4E69;
    display: flex;
    justify-content: center;
    align-items: center;
    h1 {
        font-size: 45px;
        cursor: pointer;
        a {
            text-decoration: none;
            color: #F3F4E3;
        }
    }
}