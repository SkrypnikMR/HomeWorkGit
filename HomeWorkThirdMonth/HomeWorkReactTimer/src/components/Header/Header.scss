.header{
    display: flex;
    justify-content: center;
    align-items: center ;
    color: white;
    background-color: #5C55AE;
}