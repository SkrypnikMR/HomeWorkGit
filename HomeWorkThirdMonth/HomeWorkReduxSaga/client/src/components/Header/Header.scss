.header{
    width: 100%;
    margin-top: -0;
    z-index: 1000;
    display: grid;
    justify-content:center;
    align-items: center ;
    grid-template: 1fr/ 11fr 1fr;
    color: white;
    background-color: #f5da0b;
    h1{
        text-align: left;
        padding-left:20px;
    }
    button {
        width: 250px;
        height: 50px;
        text-align: center;
        font-size: 25px;
        font-weight: bold;
        &:hover{
            background-color: #f5da0b;
        }
    }
}