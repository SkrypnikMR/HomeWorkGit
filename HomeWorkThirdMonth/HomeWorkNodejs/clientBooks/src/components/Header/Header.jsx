import React, { Component } from 'react';
import './Header.scss';
import { NavLink } from 'react-router-dom';

class Header extends Component {
    render() {
        return <header><h1> <NavLink to='/'> Books Storage</NavLink></h1></header>;
    }
}

export default Header;