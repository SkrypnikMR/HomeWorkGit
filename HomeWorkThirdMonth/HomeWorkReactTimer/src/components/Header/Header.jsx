import React, { Component } from 'react';
import './Header.scss';
import Button from '../Button';
import { ThemeContext, themes } from '../../context/themeContext';
class Header extends Component {
    render() {
        return (
            <header className="header">
                <h1>React Timer</h1>
                <ThemeContext.Consumer>
                    {context => {
                        const {toggleTheme, bulb} = context;
                        return <Button title='kek' onButtonClick={toggleTheme} disabled={false} title={bulb} />
                    }}
                </ThemeContext.Consumer>
            </header>
        )
    }
}

export default Header;