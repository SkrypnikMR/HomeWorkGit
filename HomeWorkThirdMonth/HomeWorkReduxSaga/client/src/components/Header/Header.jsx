import React, { Component } from 'react';
import './Header.scss';
import Button from '../Button';
import Modal from '../Modal';

class Header extends Component {
    constructor(props) {
        super(props);
        this.state = {
            modalState: false,
        };
    }

    handleBtnClick = () => {
        const { modalState } = this.state;
        this.setState({ modalState: !modalState });
    }

    render() {
        const { modalState } = this.state;
        return (
            <header className="header">
                <h1>Movies</h1>
                <Button onButtonClick={this.handleBtnClick} title="add Movie" />
                <Modal myClassName={modalState} onButtonClick={this.handleBtnClick} title='Add new film in collection' />
            </header>

        );
    }
}

export default Header;
