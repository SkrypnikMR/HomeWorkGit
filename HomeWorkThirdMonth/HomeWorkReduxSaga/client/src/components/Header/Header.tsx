import React, { useState } from 'react';
import './Header.scss';
import Button from '../Button';
import Modal from '../Modal';
import { useTheme } from '../hoc/withTheme.jsx';


const Header = () => {
    const [modalState, setModalState] = useState(false);
    const handleBtnClick = () :void => setModalState(!modalState);
    const { theme, setTheme } = useTheme();
    const handleThemeClick = ():void => setTheme(!theme);
    return (
        <header className="header">
            <h1>Movies</h1>
            <div className="header__controll">
                <Button
                    onButtonClick={handleThemeClick}
                    title={theme ? 'â˜€' : 'ðŸŒš'}
                />
                <Button
                    onButtonClick={handleBtnClick}
                    title="add Movie"
                />
            </div>
            <Modal
                myClassName={modalState}
                onButtonClick={handleBtnClick}
                title='Add new film in collection'
            />
        </header>

    );
}

export default Header;
