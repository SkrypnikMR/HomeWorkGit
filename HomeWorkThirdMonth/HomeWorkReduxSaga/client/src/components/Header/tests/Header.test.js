import React from 'react';
import configureStore from 'redux-mock-store';
import { shallowSmart, mountSmart } from '../../../../__tests__/helper';
import Header from '../Header';

const mockStore = configureStore();
const store = mockStore({
    movies: [],
    moviesIds: [],
    isLoading: false,
    newMovie: {},
});

describe('Header', () => {
    it('Should match snapshot', () => {
        const component = shallowSmart(<Header />, store);
        expect(component.html()).toMatchSnapshot();
    });
    it('Should have Header ', () => {
        const component = mountSmart(<Header />, store);
        expect(component.find('Header')).toHaveLength(1);
    });
    it('Should have h1 ', () => {
        const component = mountSmart(<Header />, store);
        expect(component.find('h1')).toHaveLength(1);
    });
    it('Should have Button ', () => {
        const component = mountSmart(<Header />, store);
        expect(component.find('Button')).toHaveLength(3);
    });
    it('Should have Modal ', () => {
        const component = mountSmart(<Header />, store);
        expect(component.find('Modal')).toHaveLength(1);
    });
    it('Should have Button ', () => {
        const setModalState = jest.fn();
        const component = mountSmart(<Header />, store);
        component.find('Button').at(0).props().onButtonClick();
        const handleClick = jest.spyOn(React, "useState");
        handleClick.mockImplementation(modalState => [modalState, setModalState]);
        expect(setModalState).toBeTruthy();
    });
});
