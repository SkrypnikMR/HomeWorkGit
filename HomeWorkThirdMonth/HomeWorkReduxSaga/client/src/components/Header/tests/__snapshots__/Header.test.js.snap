// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match snapshot 1`] = `"<header class=\\"header\\"><h1>React practice</h1></header>"`;
